runtime/frame_descriptors.c

```diff
@@ -46,8 +46,7 @@ static frame_descr * next_frame_descr(frame_descr * d) {
     p += num_allocs + 1;
   }
   /* Skip debug info if present */
-  if (d->frame_size & 1 &&
-      d->frame_size != (unsigned short)-1) {
+  if (d->frame_size & 1) {
```

Comment:
Is it clear that `frame_size` is not `-1` for all the frames that come before the last one?